### PR TITLE
Adds fuchsia node roles to accessibility bridge updates.

### PR DESCRIPTION
**Pull request by abrush21**
## Description

This change adds a semantics flag to mark slider UI elements, and modifies the fuchsia accessibility bridge to populate the fuchsia node role.

## Related Issues

N/A

## Tests

I added the following tests:

Accessibility bridge unit test

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [X] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation.
- [X] All existing and new tests are passing.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by chunhtai** [comment]
it looks like there are tests in framework side that grab all the possible flag from ui.SemanticsFlags and compare it with semantics property api. This create a circular dependencies.

cc @gaaclarke Do you know how should we migrate in this case?

**Comment by gaaclarke** [comment]
@abrush21 Heads up, this PR is failing because you need to run the code through the formatter.

**Comment by chunhtai** [comment]
@abrush21 We will probably need to do a three phase transition regarding the test breakage

fortunately, there are only two tests that are failing

1. create a framework pr to disable those 2 tests
    flutter/packages/flutter/test/widgets/custom_painter_test.dart: CustomPainter Supports all flags   
   flutter/packages/flutter/test/widgets/semantics_test.dart: Semantics widget supports all flags  
2. land this pr( after fixing the format error)

3. land the framework pr and enable those 2 tests again.

**Comment by abrush21** [comment]
I see that issue, and I also see a separate issue in the build_and_test_linux_unopt_debug check that Chun-Heng mentioned. A couple of framework tests (in packages/flutter/test/widgets/semantics_test.dart and flutter/packages/flutter/test/widgets/custom_painter_test.dart) are failing because of the newly introduced semantics flag. https://api.cirrus-ci.com/v1/task/5696073467428864/logs/test_framework.log. Correct me if I'm wrong, but it looks like there's a circular dependency between this change and those tests -- in order to fix the framework tests, the engine change introducing the new flag needs to land, but in order to land the engine change, the framework tests need to pass.

**Comment by abrush21** [comment]
@chunhtai Sounds good. When you say "disable" those tests, can I just comment the failing tests out in the test files? Or is there a more formal process for enabling/disabling flutter tests?

**Comment by gaaclarke** [comment]
@abrush21 Since this should only exist for a short amount of time I'd say commenting them out is fine.  Add a link to the issue and some context at the beginning of the comment.

**Comment by abrush21** [comment]
@chunhtai @gaaclarke PR 63595 (disabling failing tests) has merged, so this PR is ready for review again.

File: lib/ui/semantics.dart [hunk 1]
```diff
@@ -300,6 +300,7 @@ class SemanticsFlag {
   static const int _kIsReadOnlyIndex = 1 << 20;
   static const int _kIsFocusableIndex = 1 << 21;
   static const int _kIsLinkIndex = 1 << 22;
+  static const int _kIsSliderIndex = 1 << 23;
```
Comment (chunhtai) [inline lib/ui/semantics.dart:303]: I think we forgot to mention, we also need to update `flutter/lib/web_ui/lib/src/ui/semantics.dart`

Can you also modify the comment here to also mention this file?

Comment (chunhtai) [reply lib/ui/semantics.dart:303]: as well as /Users/chtai/git/engine/src/flutter/shell/platform/android/io/flutter/view/AccessibilityBridge.java line 1748

Comment (abrush21) [reply lib/ui/semantics.dart:303]: Sure, done.

**Comment by flar** [comment]
This test is causing failures in the engine roll (tested by running the test multiple times on this and the previous engine commit):

```
00:18 +98 ~1: /tmp/flutter sdk/packages/flutter_test/test/matchers_test.dart: matchesSemanticsData Can match all semantics flags and actions                                                           
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure object was thrown running a test:
  Expected: has semantics with actions: [
            SemanticsAction:SemanticsAction.tap,
            SemanticsAction:SemanticsAction.longPress,
            SemanticsAction:SemanticsAction.scrollLeft,
            SemanticsAction:SemanticsAction.scrollRight,
            SemanticsAction:SemanticsAction.scrollUp,
            SemanticsAction:SemanticsAction.scrollDown,
            SemanticsAction:SemanticsAction.increase,
            SemanticsAction:SemanticsAction.decrease,
            SemanticsAction:SemanticsAction.showOnScreen,
            SemanticsAction:SemanticsAction.moveCursorForwardByCharacter,
            SemanticsAction:SemanticsAction.moveCursorBackwardByCharacter,
            SemanticsAction:SemanticsAction.setSelection,
            SemanticsAction:SemanticsAction.copy,
            SemanticsAction:SemanticsAction.cut,
            SemanticsAction:SemanticsAction.paste,
            SemanticsAction:SemanticsAction.didGainAccessibilityFocus,
            SemanticsAction:SemanticsAction.didLoseAccessibilityFocus,
            SemanticsAction:SemanticsAction.customAction,
            SemanticsAction:SemanticsAction.dismiss,
            SemanticsAction:SemanticsAction.moveCursorForwardByWord,
            SemanticsAction:SemanticsAction.moveCursorBackwardByWord
          ] with flags: [
            SemanticsFlag:SemanticsFlag.hasCheckedState,
            SemanticsFlag:SemanticsFlag.isChecked,
            SemanticsFlag:SemanticsFlag.isSelected,
            SemanticsFlag:SemanticsFlag.isButton,
            SemanticsFlag:SemanticsFlag.isLink,
            SemanticsFlag:SemanticsFlag.isTextField,
            SemanticsFlag:SemanticsFlag.isReadOnly,
            SemanticsFlag:SemanticsFlag.isFocused,
            SemanticsFlag:SemanticsFlag.isFocusable,
            SemanticsFlag:SemanticsFlag.hasEnabledState,
            SemanticsFlag:SemanticsFlag.isEnabled,
            SemanticsFlag:SemanticsFlag.isInMutuallyExclusiveGroup,
            SemanticsFlag:SemanticsFlag.isHeader,
            SemanticsFlag:SemanticsFlag.isObscured,
            SemanticsFlag:SemanticsFlag.namesRoute,
            SemanticsFlag:SemanticsFlag.scopesRoute,
            SemanticsFlag:SemanticsFlag.isHidden,
            SemanticsFlag:SemanticsFlag.isImage,
            SemanticsFlag:SemanticsFlag.isLiveRegion,
            SemanticsFlag:SemanticsFlag.hasToggledState,
            SemanticsFlag:SemanticsFlag.isToggled,
            SemanticsFlag:SemanticsFlag.hasImplicitScrolling
          ] with rect: Rect.fromLTRB(0.0, 0.0, 10.0, 10.0) with size: Size(10.0, 10.0) with
elevation: 3.0 with thickness: 4.0 with platformViewId: 105 with maxValueLength: 15 with
currentValueLength: 10 with custom actions: [CustomSemanticsAction(7, label:test, hint:null,
action:null)]
  Actual: _FakeSemanticsNode:<_FakeSemanticsNode#1(Rect.fromLTRB(0.0, 0.0, 0.0, 0.0), invisible)>
   Which: flags were: [hasCheckedState, isChecked, isSelected, isButton, isTextField, isFocused,
hasEnabledState, isEnabled, isInMutuallyExclusiveGroup, isHeader, isObscured, scopesRoute,
namesRoute, isHidden, isImage, isLiveRegion, hasToggledState, isToggled, hasImplicitScrolling,
isReadOnly, isFocusable, isLink, isSlider]
When the exception was thrown, this was the stack:
#4      main.<anonymous closure>.<anonymous closure> (file:///tmp/flutter%20sdk/packages/flutter_test/test/matchers_test.dart:565:7)
#5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:146:29)
<asynchronous suspension>
#6      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart)
#7      TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:784:19)
<asynchronous suspension>
#10     TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:764:14)
#11     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1173:24)
#12     FakeAsync.run.<anonymous closure>.<anonymous closure> (package:fake_async/fake_async.dart:178:54)
#17     withClock (package:clock/src/default.dart:48:10)
#18     FakeAsync.run.<anonymous closure> (package:fake_async/fake_async.dart:178:22)
#23     FakeAsync.run (package:fake_async/fake_async.dart:178:7)
#24     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1170:15)
#25     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:138:24)
#26     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:175:19)
<asynchronous suspension>
#27     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart)
#32     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:173:13)
#33     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:231:15)
#38     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:228:5)
#39     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:383:17)
<asynchronous suspension>
#40     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart)
#45     Invoker._onRun.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:370:9)
#46     Invoker._guardIfGuarded (package:test_api/src/backend/invoker.dart:415:15)
#47     Invoker._onRun.<anonymous closure> (package:test_api/src/backend/invoker.dart:369:7)
#54     Invoker._onRun (package:test_api/src/backend/invoker.dart:368:11)
#55     LiveTestController.run (package:test_api/src/backend/live_test_controller.dart:153:11)
#56     RemoteListener._runLiveTest.<anonymous closure> (package:test_api/src/remote_listener.dart:256:16)
#61     RemoteListener._runLiveTest (package:test_api/src/remote_listener.dart:255:5)
#62     RemoteListener._serializeTest.<anonymous closure> (package:test_api/src/remote_listener.dart:208:7)
#80     _GuaranteeSink.add (package:stream_channel/src/guarantee_channel.dart:125:12)
#81     new _MultiChannel.<anonymous closure> (package:stream_channel/src/multi_channel.dart:159:31)
#85     CastStreamSubscription._onData (dart:_internal/async_cast.dart:85:11)
#119    new _WebSocketImpl._fromSocket.<anonymous closure> (dart:_http/websocket_impl.dart:1145:21)
#127    _WebSocketProtocolTransformer._messageFrameEnd (dart:_http/websocket_impl.dart:338:23)
#128    _WebSocketProtocolTransformer.add (dart:_http/websocket_impl.dart:232:46)
#138    _Socket._onData (dart:io-patch/socket_patch.dart:2044:41)
#147    new _RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1580:33)
#148    _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:1076:14)
(elided 111 frames from dart:async and package:stack_trace)
This was caught by the test expectation on the following line:
  file:///tmp/flutter%20sdk/packages/flutter_test/test/matchers_test.dart line 565
The test description was:
  Can match all semantics flags and actions
```

**Comment by chunhtai** [comment]
Hi @flar, thanks for reverting the pr. That is a valid test failure caused by this PR. Do you know where can we get the full list of the test failures of the engine roll?

**Comment by flar** [comment]
I didn't see (or don't remember seeing) any other failures that I thought would come from this PR.

**Comment by flar** [comment]
You can check the test results from any of the engine PRs that "mentioned this PR" before my revert...

**Comment by chunhtai** [comment]
@flar Thanks!

@abrush21 , it seems like there is one more test you need to disable in the framework in order to merge this PR https://github.com/flutter/flutter/blob/2f96d1f0263dfdaf9f1c8d4f0fa054839342e350/packages/flutter_test/test/matchers_test.dart#L529

**Comment by abrush21** [comment]
Sounds good, created pull request https://github.com/flutter/flutter/pull/65194.

**Pull request merged**